src/main/java/use_cases/sales_report/SalesReporter.java

```diff
@@ -7,32 +7,46 @@
 import java.util.List;
 
 
-public class SalesReporter {
-    private final String FILE_PATH = "C:\\Users\\leste\\OneDrive\\Documents\\AutoCAD Sheet Sets"
+public class SalesReporter implements SalesReporterInputBoundary {
+    private final SalesReporterOutputBoundary presenter;
+    /*private final ShowHistoryDsGateway checker;*/
     private final int DATE_TIME_COLUMN = 0;
     private final int ACTION_COLUMN = 2;
     private final int QUANTITY_COLUMN =  4;
     private final int SERIAL_NUM_COLUMN = 5;
 
-    private final String[] HEADERS =
-            {"Item Serial No.", "Item Name", "Price", "Quantity Sold", "Items Returned", "Revenue"};
+    public SalesReporter(SalesReporterOutputBoundary presenter, ShowHistoryDsGateway checker) {
+        this.presenter = presenter;
+        this.checker = checker;
+    }
+
     // return a list of string arrays representing where each element in the list is a row in the sales report, and each
     // element in the array is a column of that row.
-    public ArrayList<String[]> generateReport(LocalDateTime startTime, LocalDateTime endTime){
-        ArrayList<String[]> splitData = splitListTimeRange(rows, startTime, endTime);
-        ArrayList<String> serialNumList = serialNumList(splitData);
-        ArrayList<String[]> result = new ArrayList<String[]>();
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
-        String formattedStartTime = startTime.format(formatter);
-        String formattedEndTime = endTime.format(formatter);
-        String dateRange = "Time Range: " + formattedStartTime + " - " + formattedEndTime;
-        result.add(new String[] {dateRange});
-        result.add(HEADERS);
-        for (String serialNum: serialNumList){
-            result.add(getRow(splitData, serialNum));
+    public void generateReport(List<String[]> rows, String startTime, String endTime){
+        LocalDateTime start = stringToDateTime(startTime);
+        LocalDateTime end = stringToDateTime(endTime);
+        if (!checker.StartDateValid(start)){
+            presenter.prepareFailure(SalesReporterOutputBoundary.START_TIME_ERROR);
         }
-        String totalRevenue = String.format("Total Revenue: $%.5f", totalRevenue(splitData));
-        result.add(new String[]{totalRevenue});
+        else if (!checker.EndDateValid(end)){
+            presenter.prepareFailure(SalesReporterOutputBoundary.END_TIME_ERROR);
```

Comment:
I can't seem to find the SalesReporterOutputBoundary interface or the presenter... not sure what the reason is though.

Reply:
It might be that I'm just not on the most recent version.

Reply:
Yeah i dont know why it wasn't pushed. SalesReporterView, SalesReporterOutputBoundary, and SalesReporterPresenter are all highlighted red in my project.
![image](https://user-images.githubusercontent.com/113144479/205463110-d79cb395-e21b-4e14-98ed-1114158780a1.png)

not sure why